### PR TITLE
Add integration test for ForEachTask

### PR DESCRIPTION
**Pull request by hectorcast-db**
## Changes
Add integration test for ForEachTask

## Tests
- [x] `make test` run locally
- [ ] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Review by tanmay-db — APPROVED** [review]

File: internal/acceptance/job_test.go [hunk 1]
```diff
@@ -110,6 +110,71 @@ func TestAccJobTasks(t *testing.T) {
...
+					node_type_id  = data.databricks_node_type.smallest.id
+				}
+			}
+
+			task {
+				task_key = "for_each_task_key"
+				for_each_task {
+					concurrency = 1
+					inputs = "[1, 2, 3, 4, 5, 6]"
+					task {
```
Comment (tanmay-db) [inline internal/acceptance/job_test.go:150]: Might be good to have another task in the `for_each_task`.

File: internal/acceptance/job_test.go [hunk 2]
```diff
@@ -110,6 +110,71 @@ func TestAccJobTasks(t *testing.T) {
 	})
 }
 
+func TestAccForEachTask(t *testing.T) {
+	t.Skip("Skipping this test because feature not enabled in Prod")
```
Comment (tanmay-db) [inline internal/acceptance/job_test.go:114]: cc: @geekswaroop

Comment (geekswaroop) [reply internal/acceptance/job_test.go:114]: Acknowledged.

File: internal/acceptance/job_test.go (hunk 1 above)
Comment (tanmay-db) [reply internal/acceptance/job_test.go:150]: Ah got it.

File: internal/acceptance/job_test.go [hunk 3]
```diff
@@ -110,6 +110,71 @@ func TestAccJobTasks(t *testing.T) {
...
+			job_cluster {
+				job_cluster_key = "j"
+				new_cluster {
+					num_workers   = 20
+					spark_version = data.databricks_spark_version.latest.id
+					node_type_id  = data.databricks_node_type.smallest.id
+				}
+			}
+
+			task {
```
Comment (geekswaroop) [inline internal/acceptance/job_test.go:145]: `task` is part of API 2.0. Can we also have a test creating a multi task job? A job containing a `notebook_task` and a `for_each_task` should be fine.

**Review by geekswaroop — APPROVED** [review]
An API 2.1 test would be nice, otherwise LGTM

File: internal/acceptance/job_test.go [hunk 4]
```diff
@@ -110,6 +110,71 @@ func TestAccJobTasks(t *testing.T) {
...
+			name = "{var.RANDOM}"
+
+			job_cluster {
+				job_cluster_key = "j"
+				new_cluster {
+					num_workers   = 20
+					spark_version = data.databricks_spark_version.latest.id
+					node_type_id  = data.databricks_node_type.smallest.id
+				}
+			}
```
Comment (alexott) [inline internal/acceptance/job_test.go:143]: Why do we need a job cluster if it's not used?

File: internal/acceptance/job_test.go (hunk 3 above)
Comment (alexott) [reply internal/acceptance/job_test.go:145]: No, the `task` block is for API 2.1

**Review by alexott — APPROVED** [review]

**Pull request merged**